src/spikeinterface/core/generate.py

```diff
@@ -566,6 +566,116 @@ def synthetize_spike_train_bad_isi(duration, baseline_rate, num_violations, viol
     return spike_train
 
 
+from spikeinterface.core.basesorting import BaseSortingSegment, BaseSorting
+
+
+class SortingGenerator(BaseSorting):
+    def __init__(
+        self,
+        num_units: int = 5,
+        sampling_frequency: float = 30000.0,  # in Hz
+        durations: List[float] = [10.325, 3.5],  #  in s for 2 segments
+        firing_rates: float | np.ndarray = 3.0,
+        refractory_period_ms: float | np.ndarray = 4.0,  # in ms
+        seed: Optional[int] = None,
+    ):
+        unit_ids = np.arange(num_units)
```

Comment:
Will we have a docstring for this? For example you have durations as a list of 2 floats for 2 segments, so for future users it might be beneficial for them to know that the len(durations)==n_segments. And what are the implications of giving one firing rate vs an array of firing rates if I as a user try something like this?

Reply:
Yeah, we should. The duration thing is a very pervasive in a bunch of functions we should document it. Less pevasive but not exclusive to this one is the firing rates as an array or float.

Reply:
@h-mayorquin can you add a docstring here? Maybe explaining the difference with the `generate_sorting`

Reply:
Added.

Reply:
I also corrected a typo of refactory to refractory